WallsAndHoles/newmapdialog.cpp

```diff
@@ -1,94 +1,79 @@
 #include "newmapdialog.h"
 
+#include <QGridLayout>
+#include <QVBoxLayout>
+#include <QGroupBox>
+#include <QLabel>
+#include <QDialogButtonBox>
+#include <QHBoxLayout>
+
 NewMapDialog::NewMapDialog(QDialog *parent)
-  : QDialog(parent),
-  result(NewMapData(-1,-1))
+    : QDialog(parent)
 {
     //Setup the dialog
     QDialog::resize(400,200);
     QDialog::setWindowTitle("New Map");
     QDialog::setModal(true);
 
-    dialogBtnBox = new QDialogButtonBox(this);
-    dialogBtnBox->addButton("OK",QDialogButtonBox::AcceptRole);
-    dialogBtnBox->addButton("Cancel",QDialogButtonBox::RejectRole);
-
-    setupSpinBox();
-    setupCheckBox();
-    setMainLayout();
-
-
-    //Connections for the OK and Cancel buttons
-    connect(dialogBtnBox, &QDialogButtonBox::accepted, this, &NewMapDialog::setResult);
-    connect(dialogBtnBox, &QDialogButtonBox::rejected, this, &NewMapDialog::cancelled);
-
-    //Connections for the check boxes
-    TileMapProperty.outdoorMap = outdoorMapChkBox->isChecked();
-    connect(outdoorMapChkBox, &QCheckBox::toggled, this,
```

Comment:
No need to change any values every time they change. We only care about the values once the dialog is done